### ISSUE TITLE
[BUG] the app is reducing the path of the url to lowercase, which is against RFC specs

### ISSUE THREAD

**Issue by sapiency** [icefields/Power-Ampache-2#145]
hi,
now I should be in the right place ...

please think about what you are doing with urls!
everything between http*:// and / is not case sensitive, everything after '/' is case sensitive.
I could not login to my Nextcloud server and so in the log of the webserver, that you put the whole string
"Server URL" to lower case ...
May you verify this please?

BR

**Comment by icefields** [comment]
thanks for reporting, I will take a look and eventually fix

**Comment by icefields** [comment]
While I debug the issue, 
please make sure you are you using the latest Nextcloud Music plugin version (vers > 2.0).
There is a bug in versions prior to 2.0 realated to mixed-case Ampache usernames, that has been fixed in 2.0

**Comment by sapiency** [comment]
thanks for the hint, I've seen this in the app. But I'm a little bit confused ...
vers > 2.0 means that it is not accepting 2.0 ... but this is the latest version
maybe it is better to write 
vers >= 2.0 
BR

**Comment by icefields** [comment]
yes that is correct, I made a mistake. Please install Nextcloud Music version >= 2 and let me know if that solves the issue.

**Comment by sapiency** [comment]
later ... i can only try if I do not see the 404 in the log of the webserver ....

"GET /xxxxxx/server/json.server.php?action=ping HTTP/1.1" 404 3079 "-" "PowerAmpache2-1.00-64-fdroid"
should look like:
"GET /xxXxxX/server/json.server.php?action=ping HTTP/1.1" 200 3079 "-" "PowerAmpache2-1.00-64-fdroid"

and this "xxxxxx" ist what I've asked you to look for ...

BR

**Comment by icefields** [comment]
Yes, I understand your initial description. The path and query components of a URL should indeed be case-sensitive. However, before addressing that, it's important to ensure you're using the correct version of the Music plugin, which is why I was asking.

To be more clear, it's better to use technical terms when reporting a bug, you showed me some examples, but didn't tell me if `xxxxx` is the host or the path, I'm assuming is the path because I recognize the logs. 
According to the standards defined in RFC 3986, which defines the case sensitivity of different URL components, if we take the following URL as an example `https://www.Example.com/Path/to/Resource?param=value#Fragment`, we can divide it into the following components:
**Scheme**: https (case-insensitive)
**Host**: www.Example.com (case-insensitive)
**Path**: /Path/to/Resource (case-sensitive)
**Query String**: param=value (case-sensitive depending on server)
**Fragment**: Fragment (case-sensitive)

It appears that the Path is being converted to lowercase, which should be case-sensitive. This is indeed a bug, and I will fix it in the next release. Unfortunately, it's too late to include this fix in the current release, as it is being finalized today.

**Comment by sapiency** [comment]
thanks,
hmm, and also thanks for looking on the RFC and also for the spec, I didn't thought about to do this. I looked in the logfile (first nextcloud, then werbserver) found the  path in lower case, checkt out the code, looked for the string and finaly grep told me that I should look in the apk files if I want to see more details ... to much effort ;-). And yes, it is better to use technical terms, but it is also good to know that if the response of the webserver on a "GET" request ist 404, the application behind does not see anything of the request. 
BR and lets wait for the next version ..

**Comment by icefields** [comment]
no problem at all, I just want to be precise so I don't waste time working on the wrong thing.

Anyway, I found the issue, it's a regression bug from when I started working on the multi-user feature and decided to use the server-url as part of the identifier, reducing it to lowercase in the process. That's clearly a mistake, I already started working on the fix, which will be available in the next version.
Are you using the app from playstore, fdroid or github-releases?

**Comment by sapiency** [comment]
see post with log messages above ... fdroid ....

**Comment by icefields** [comment]
ok, you're right. If you are ok to do test, I'm going to send you a link for an apk with a fix I just made. In a minute, I'm creating the fix-branch now

**Comment by icefields** [comment]
If you're ok with testing the patch, this is the apk with the fix, please let me know if the bug is fixed. The fixes are in the branch called `bug/145-url-path-lowercase` , I will create a pr and merge once you confirm it's working for you.
https://github.com/icefields/Power-Ampache-2/raw/bug/145-url-path-lowercase/app/Github/release/app-Github-release.apk
I created a github version release because I can't sign FDroid apps.
Thanks.

**Comment by sapiency** [comment]
ok, I've testet the app with waydroid, found one mistake from my side because I didn't use the right URL to the app. But now I can see, that the path is case sensitive, so this problem is fixed.
On waydroid I get now the error "cannot parse time". Could this be a side effect?
(I've compared the log entries between the app from the store and your apk, where I still get the old response (log + app) )

**Comment by icefields** [comment]
hello, thanks for testing, I use Waydroid myself on Arch all the time, and I can say it's not reliable for testing. I don't know what "cannot parse time" issue is exactly, but if I had to guess, it sounds like it's related to the clock of waydroid not in sync with your server, so the client cannot get authorized. 
Can you test on an actual device? If you can do that, and you can confirm the problem is fixed, I will push the fix in the next release, which will happen this weekend.

**Comment by sapiency** [comment]
hi,
sorry I've currently no device to test apps from outside of the store.
Time:
I thought about this, but didn't expected a problem out of not synchronized time. But I looked at the settings, modified them from auto to manual and selected the local stuff ... and finaylly I could log in ;-) ... on waydroid ... and I saw the songs I've already uploaded and could play them. So it looks like that it works. Thanks
BR
